File: server/Infrastructure/LobTools/Models/CommonModels.cs
namespace Brainvest.Dscribe.LobTools.Models;

using System.Collections.Generic;

public class LobSummaryInfo
{
	public int CommentsCount { get; set; }
	public int AttachmentsCount { get; set; }
}

public class LobSummaryResponse
{
	public string EntityTypeName { get; set; }
	public Dictionary<int, LobSummaryInfo> Summaries { get; set; }
}

public class LobSummaryRequest
{
	public IEnumerable<int> Identifiers { get; set; }
	public string EntityTypeName { get; set; }
}

File: client/src/lib/common/models/lob/common-models.ts
export interface LobSummaryInfo {
	AttachmentsCount: number;
	CommentsCount: number;
}

export class LobSummaryResponse {
	EntityTypeName: string;
	Summaries: { [key: number]: LobSummaryInfo };
}

export class LobSummaryRequest {
	constructor(public EntityTypeName: string, public Identifiers: number[]) {
	}
}
